src/hb-ot-var-common.hh

```diff
@@ -416,6 +418,45 @@ struct TupleVariationHeader
   DEFINE_SIZE_MIN (4);
 };
 
+/* not using hb_bytes_t: avoid potential build issues with some compilers */
+struct byte_data_t
+{
+  const char *p = nullptr; //allocated by hb_calloc () or similar
+  unsigned length = 0;
```

Comment:
Could this instead just hold an instance of hb_bytes_t and forward the needed methods to it? That way we don't need to duplicate the functionality of hb_bytes_t. I think that shouldn't trigger the compilation issues since it won't have an overloaded & operator.